### ISSUE TITLE
"Too many redirects when fetching" and unable to download report

### ISSUE THREAD

**Issue by Smithstorian** [philipmulcahy/azad#204]
Hi Philip - I didn't realize this was a place to submit issues and wrote something about this on the chrome extension site. Sorry for duplication. And first I want to thank you for this great tool.

OK, so I've been using this extension for a while and am now hitting this issue this year. I am getting an error when the extension starts scraping "HTTP0: too many redirects when fetching https://amazon.com:443/gpredirect.html ...." I uploaded an image of the error. I've tried different years and also have cleared cache (through the extension) and logged out and back in (both through the extension and manually) and have also shut down and restarted my PC. 

Sometimes I end up with an online view of the report but the download csv button doesn't do anything. 

Am I doing something wrong here? Thanks for everything! -Shannon
![AZAD Issue](https://user-images.githubusercontent.com/126113596/220754407-55a623df-8662-4237-83e0-81a851485b42.JPG)

**Comment by svhano** [comment]
Same issue here. Seems to be the audible part. four lines popping up showing the same content "bla...audible....bla" and then simply nothing shows up.
I love your extension, it does exactly what I was looking for. 
But in my case the year 2021 never shows the csv-result.

**Comment by philipmulcahy** [comment]
I don't think I have any audible content in my 3 amazon accounts (I have one for myself in .co.uk, and then one in each of .fr and .com so as to help with testing this extension), so this is going to be tricky to debug. 

One memory the error message you attached triggers is the partial logout thing. Sometimes bits of Amazon log you out without telling you. Azad is making queries to several bits of their URL space, and sometimes the logged out bits send HTTP response codes that confuse the javascript URL fetcher into an infinite loop. Have you tried forcibly logging yourselves out (there's a button on the extensions popup control window that does a good job of forcing logout)? It's a longshot, because I know that I've not had any audible orders to play with, but I think worth a try.

**Comment by poken1151** [comment]
Appreciate the suggestion, unfortunately doesn't help. 

When I get hit with the redirect error, it seems to be taking me to a specific order. Also, if it is this specific order, I doubt you'll be able to recreate it without buying their annual membership.

The too many redirects occurs when:
1. I have an order history with the 12month subscription.
2. I have the box checked for "Show Items not Orders".

When I do that, it seems to just load to a white page with the error flashed above.

I have tried:

1. Logging out via the extension's button.
2. Logging out direcly.
3. Logging out of Audible.
4. Logging in to Audible manually before running.
5. Running the extension without Logging into Audible.
6. Repeating various combinations of the above across three browsers.
7. Repeating various combinations of the above across all incognito/inPrivate modes.

Is it possible to include buttons to exclude some sites if the bug can't be fixed due to access? Or, any additional info from developer mode or options we can provide to help resolve?

**Comment by poken1151** [comment]
Quick update: As a temporary fix, I was able to Archive the order from my main amazon order history page. This seemed to fix the issue by hiding it from the extension. So now selecting the Show Items option and running no longer resulted in an error and I can finally get the detailed CSV. 

Still, would suggest a fix, will fork and see if I can help.

**Comment by philipmulcahy** [comment]
I've got an idea in flight that might make azad more robust in the scenario where it gets stuck on one page. It might offer a solution of sorts to many outstanding problems users have reported - the interesting bit now (for me) is how to report the problems the extension has skipped past, because otherwise I may get deluged in "thanks for the (partial) results - where are my $new_amazon_product_type orders?"

**Comment by philipmulcahy** [comment]
from the chrome web store support page:
```
Hey Philip - Thank you for this great tool. I know this is just a part-time thing for you and here you are doing more than a gazillion-dollar global corporation. I'm suddenly having an issue getting "HTTP0: too many redirects when fetching https://www.amazon.com:443/gp/redirect.html/......." I can send you a copy of the issue. I do sometimes get a report that shows online but am never able to download it. When I click on download nothing happens. I tried on different years and have cleared cache (through your extension) and logged out and back in and even shut down and restarted my pc. Any thoughts? And thanks again you are appreciated. -Shannon
```

**Comment by philipmulcahy** [comment]
1.9.34 has some concurrency fixes that are targeted at the described scenario.
They aim to:
1) be able to get past orders that it doesn't know how to read
2) log some info to the debug console, hopefully helping us fix the individual orders

please feel free to re-open if it's still happening, preferably with an attached console log.

**Comment by NodeGuy** [comment]
I'm a new user and this is happening to me.

[www.amazon.com-1710447206740.log](https://github.com/philipmulcahy/azad/files/14607477/www.amazon.com-1710447206740.log)

**Comment by Smithstorian** [comment]
Hey all - it happened to me again this year as I grabbed all my transactions for tax purposes. It does seem to have one specific place that it just stops in and then I get the fetch error. I still appreciate this tool!

**Comment by NodeGuy** [comment]
> I'm a new user and this is happening to me.
> 
> [www.amazon.com-1710447206740.log](https://github.com/philipmulcahy/azad/files/14607477/www.amazon.com-1710447206740.log)

I fixed the bug that was affecting me in pull request #270.

**Comment by philipmulcahy** [comment]
I'm going to look at this and the PR this weekend. Thank you everyone!

**Comment by philipmulcahy** [comment]
released in 1.12.0
